mslib/utils/mssautoplot.py

```diff
@@ -24,12 +24,316 @@
+    try:
+        doc = xml.dom.minidom.parse(datasource)
+    except xml.parsers.expat.ExpatError as ex:
+        raise SyntaxError(str(ex))
+
+    ft_el = doc.getElementsByTagName("FlightTrack")[0]
+
+    waypoints_list1 = []
+    waypoints_list2 = []
+    for wp_el in ft_el.getElementsByTagName("Waypoint"):
+
+        location = wp_el.getAttribute("location")
+        lat = float(wp_el.getAttribute("lat"))
+        lon = float(wp_el.getAttribute("lon"))
+        flightlevel = float(wp_el.getAttribute("flightlevel"))
+        comments = wp_el.getElementsByTagName("Comments")[0]
+        # If num of comments is 0(null comment), then return ''
+        if len(comments.childNodes):
+            comments = comments.childNodes[0].data.strip()
+        else:
+            comments = ''
+
+        waypoints_list1.append((lat, lon, flightlevel, location, comments))
+        waypoints_list2.append(ft.Waypoint(lat, lon, flightlevel, location, comments))
+        waypoints_list2[-1].utc_time = datetime.now()
+    return waypoints_list1, waypoints_list2
+
+
+class Plotting():
+    def __init__(self, cpath):
+        read_config_file(cpath)
+        self.config = config_loader()
+        self.num_interpolation_points = self.config["num_interpolation_points"]
+        self.num_labels = self.config["num_labels"]
+        self.tick_index_step = self.num_interpolation_points // self.num_labels
+        self.wps = None
+        self.bbox = None
+        self.wms_cache = config_loader(dataset="wms_cache")
+        self.fig = plt.figure()
+        section = self.config["automated_plotting_flights"][0][1]
+        filename = self.config["automated_plotting_flights"][0][3]
+        self.params = mslib.utils.coordinate.get_projection_params(
+            self.config["predefined_map_sections"][section]["CRS"].lower())
+        self.params["basemap"].update(self.config["predefined_map_sections"][section]["map"])
+        self.bbox_units = self.params["bbox"]
+        self.wpslist = []
+        self.wpslist = load_from_ftml(filename)
+        self.wps = self.wpslist[0]
+        self.wp_model_data = self.wpslist[1]
+        self.wp_lats, self.wp_lons, self.wp_locs = [[x[i] for x in self.wps] for i in [0, 1, 3]]
+        self.wp_press = [mslib.utils.thermolib.flightlevel2pressure(wp[2] * units.hft).to("Pa").m for wp in self.wps]
+        self.fig.clear()
+        self.ax = self.fig.add_subplot(111, zorder=99)
+        self.path = [(wp[0], wp[1], datetime.now()) for wp in self.wps]
+        self.vertices = [list(a) for a in (zip(self.wp_lons, self.wp_lats))]
+        self.lats, self.lons = mslib.utils.coordinate.path_points([_x[0] for _x in self.path],
+                                                                  [_x[1] for _x in self.path],
+                                                                  numpoints=self.num_interpolation_points + 1,
+                                                                  connection="greatcircle")
+
```

Comment:
we could also think about moving path, draw, setup method to the Plotting class,

https://www.techversantinfotech.com/python-naming-conventions-points-you-should-know/#:~:text=Method%20name%20in%20Python,it%20needs%20to%20be%20mangled.

Reply:
yes. I focused here mostly on the mslib/msui stuff and left the CLI tool mostly untouched except for moving it to a single file..